### PR TITLE
Various small fixes and improvements to the glossary

### PR DESCRIPTION
**Pull request by saivann**
Here is a bunch of small fixes and suggestions (please feel free to nitpick what you think deserves to be merged)

Add link to the glossary from each definition
Fix missing BIP37 link
Fix searchbox width on mobiles
Apply more consistent margin, padding and text-align style for the searchbox
Use H3 titles in the glossary_entry layout for better readability
Always use HTML classes for CSS stylesheets
Fix one typo in glossary.rb

### CODE REVIEW

File: _less/screen.less [hunk 1]
```diff
@@ -907,9 +907,11 @@ table td,table th{
 	border:2px solid #ee9209;
 	color:#fff;
 }
-#glossary_term {
+.glossary_term {
 	display: block;
 	width: 20em;
+	text-align: center;
```
Comment (harding) [inline _less/screen.less]: This seems to center the text in the search box, which looks weird to me.  I think Google and every other search box I've ever used was left-aligned.

2015-04-09-144534_1280x800_scrot

How do you feel about me removing this one line when I merge?

Comment (saivann) [inline _less/screen.less]: This was made on purpose due to the searchbar being centered and trapped between centered texts, but sure, if you find it weird, please feel free to drop it. If you'd ask me, a centered searchbox seemed usual to me but now that you're mentionning it, I can't find a good example at hands :)

**Comment by harding** [comment]
Merged as a790f353d0b87a7326926e56492accf94740d25b with the one change we disucessed.  Thanks!

**Comment by saivann** [comment]
Thanks!

**Pull request closed without merging**